### PR TITLE
Update to readme for cse

### LINKED ISSUE #151: Add a client-side encryption section to the README
* Add a table explaining the features of the different encryption modes supported.
* Add documentation on how to enable client-side encryption.
* Add note about IV attacks if a single key is used to encrypt 6 million+ objects.

### PR DESCRIPTION
**Pull request by geek**
Closes #151

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,6 +17,12 @@ SDK for interacting with Joyent's Manta system.
 ### Requirements
 * [Java 1.8](http://www.oracle.com/technetwork/java/javase/downloads/index.html) or higher.
 * [Maven 3.3.x](https://maven.apache.org/)
+* [Java Cryptography Extension](http://www.oracle.com/technetwork/java/javase/downloads/jce8-download-2133166.html)
+
+#### Add [BouncyCastle](http://www.bouncycastle.org/latest_releases.html) as a security provider
```
Comment (dekobon) [inline README.md]: Why is this needed? If there is a problem, we should fix it in the code.

Comment (geek) [inline README.md]: After installing JSE I had to also add bouncy castle to the list of trusted providers

Comment (dekobon) [inline README.md]: Was this for using the CLI?

Comment (geek) [inline README.md]: Yes

Comment (dekobon) [inline README.md]: Yeah, I would add a note saying that is for the CLI and not for the core library.

**Comment by geek** [comment]
@dekobon fixed
